### PR TITLE
[qosorch]: Add possibility to set min value for WRED thresholds.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
- Set min thresholds equal to max to preserve default behavior.

### CODE REVIEW

File: orchagent/qosorch.cpp [hunk 1]
```diff
@@ -284,32 +285,95 @@ bool WredMapHandler::convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tupl
             attr.value.s32 = stoi(fvValue(*i));
             attribs.push_back(attr);
 
-            // set min threshold to the same value as MAX
-            attr.id = SAI_WRED_ATTR_YELLOW_MIN_THRESHOLD;
-            attr.value.s32 = stoi(fvValue(*i));
-            attribs.push_back(attr);
+            auto min_thresh = find_if(attribs.begin(), attribs.end(),
```
Comment (stcheng) [inline orchagent/qosorch.cpp]: why this is also min_thresh as the attribute is max threshold?

File: orchagent/qosorch.cpp [hunk 2]
```diff
@@ -284,32 +285,95 @@ bool WredMapHandler::convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tupl
...
         {
             attr.id = SAI_WRED_ATTR_GREEN_MAX_THRESHOLD;
             attr.value.s32 = stoi(fvValue(*i));
             attribs.push_back(attr);
 
-            // set min threshold to the same value as MAX
-            attr.id = SAI_WRED_ATTR_GREEN_MIN_THRESHOLD;
-            attr.value.s32 = stoi(fvValue(*i));
-            attribs.push_back(attr);
+            auto min_thresh = find_if(attribs.begin(), attribs.end(),
```
Comment (stcheng) [inline orchagent/qosorch.cpp]: same question as above

File: orchagent/qosorch.cpp [hunk 3]
```diff
@@ -284,32 +285,95 @@ bool WredMapHandler::convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tupl
...
         {
             attr.id = SAI_WRED_ATTR_RED_MAX_THRESHOLD;
             attr.value.s32 = stoi(fvValue(*i));
             attribs.push_back(attr);
 
-            // set min threshold to the same value as MAX
-            attr.id = SAI_WRED_ATTR_RED_MIN_THRESHOLD;
-            attr.value.s32 = stoi(fvValue(*i));
-            attribs.push_back(attr);
+            auto min_thresh = find_if(attribs.begin(), attribs.end(),
```
Comment (stcheng) [inline orchagent/qosorch.cpp]: same as above

File: orchagent/qosorch.cpp [hunk 4]
```diff
@@ -284,32 +285,95 @@ bool WredMapHandler::convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tupl
...
+            {
+                // set min threshold to the same value as MAX
+                attr.id = SAI_WRED_ATTR_RED_MIN_THRESHOLD;
+                attr.value.s32 = stoi(fvValue(*i));
+                attribs.push_back(attr);
+            }
+        }
+        else if (fvField(*i) == red_min_threshold_field_name)
+        {
+           auto min_thresh = find_if(attribs.begin(), attribs.end(),
```
Comment (stcheng) [inline orchagent/qosorch.cpp]: why do we want to find the min attribute here? why not set it directly similar to line 348-351?

File: orchagent/qosorch.cpp (hunk 1 above)
Comment (stcheng) [inline orchagent/qosorch.cpp]: Add comments here: when min threshold is not found, use the same value as max.

**Comment by stcheng** [comment]
i'm wondering if it is better to enforce the user to put both values into the config file. then we will only need to check if the values exist and the min value is less than or equal to the max value.

File: orchagent/qosorch.cpp (hunk 4 above)
Comment (oleksandrivantsiv) [inline orchagent/qosorch.cpp]: We don't know in which order we are getting min and max values. If we got max value before min we should change default. Anyway I'm going to add min as required value and will change validation logic

**Comment by oleksandrivantsiv** [comment]
@stcheng I've done experiments and found that the best way to validate thresholds is to pass them to SAI redis. It already contains validation and track dependencies between attributes.

**Pull request merged**